subsys/net/lib/sockets/sockets.c

```diff
@@ -129,6 +129,12 @@ int _impl_zsock_socket(int family, int type, int proto)
 	}
 #endif
 
+#if defined(CONFIG_NET_SOCKETS_CAN)
+	if (family == AF_CAN && type == SOCK_RAW) {
```

Comment:
This is the same problem as with initial packet socket patch. Please dispatch the entire address family to your implementation, and deal with different socket types within it there. Please fix this in https://github.com/zephyrproject-rtos/zephyr/pull/12808

Reply:
I have actually a new PR coming that will remove all these settings from sockets.c. So socket families will just register their stuff in their individual files and sockets.c will use them automatically. I will send the PR next week.

Reply:
Cool. We'll have it under review. In the meantime, please fix this issue in https://github.com/zephyrproject-rtos/zephyr/pull/13168